spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -125,6 +125,26 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("uint data type support") {
+    Seq(true, false).foreach { dictionaryEnabled =>
+      Seq(Byte.MaxValue, Short.MaxValue).foreach { valueRanges =>
+        {
+          withTempDir { dir =>
+            val path = new Path(dir.toURI.toString, "testuint.parquet")
+            makeParquetFileAllTypes(path, dictionaryEnabled = dictionaryEnabled, valueRanges + 1)
+            withParquetTable(path.toString, "tbl") {
+              if (CometSparkSessionExtensions.isComplexTypeReaderEnabled(conf)) {
+                checkSparkAnswer("select _9, _10 FROM tbl order by _11")
```

Comment:
Do we already have logic to fall back to Spark when the complex type reader is enabled?

Reply:
No we don't for two reasons. Firstly, in the plan we get the schema as understood by Spark so all the _signed_ int_8 and int_16 values are indistinguishable from the unsigned ones. As a result we fall back to Spark for both signed and unsigned integers. Secondly, too many unit tests fail because we check that the plan contains a comet operator and would need to be modified. 
I'm open to putting it back though.

Reply:
> As a result we fall back to Spark for both signed and unsigned integers.

Just 8 and 16 bit, or all integers? I'm fine with falling back for 8 and 16 bit for now, although it would be nice to have a config to override this (with the understanding that behavior is incorrect for unsigned integers).

Reply:
Just 8 and 16 bit. 
I started with the fallback to spark _and_ a compat override. The reason I reverted it is that I couldn't see a way to get to compatibility with spark even after/if https://github.com/apache/arrow-rs/issues/7040 is addressed. 
Let me do as you suggest. Marking this as draft in the meantime.